docs: document sync/--no-sync option and disk-durable publisher confirms (#2120)
cloudamqp/lavinmq: docs/configuration.md

| `tcp_recv_buffer_size` | — | — | Int | (system) | TCP receive buffer size |
| `tcp_send_buffer_size` | — | — | Int | (system) | TCP send buffer size |
| `segment_size` | — | — | Int | `8388608` | Message store segment size (bytes, 8MB) |

> | `sync` | `--no-sync` | `LAVINMQ_SYNC` | Bool | `true` | Flush pending writes to the data directory with `syncfs(2)` before acking publisher confirms (`Basic.Ack`) and transaction commits, so confirmed messages survive sudden power loss or a kernel crash. Disabling (`--no-sync` / `sync = false`) leaves durability to the OS — unsafe for production, but speeds up CI and local development. See [Publisher Confirms](publisher-confirms.md#disk-durable-confirms). |

Reviewer: leaving it to OS is what rabbitmq classic/stream queues does, kafka etc. So i'm not sure it's "unsafe", just other guarantees. It's only really unsafe if single node, or all other nodes exists in the same power domain (ie they can have a power cut at the exact same time). 

```suggestion
| `sync` | `--no-sync` | `LAVINMQ_SYNC` | Bool | `true` | Flush pending writes to the data directory with `syncfs(2)` before acking publisher confirms (`Basic.Ack`) and transaction commits, so confirmed messages survive sudden power loss or a kernel crash. Disabling (`--no-sync` / `sync = false`) leaves durability to the OS and its page cache. See [Publisher Confirms](publisher-confirms.md#disk-durable-confirms). |
```
